test/backend/datastores.mock.spec.js

```diff
@@ -0,0 +1,183 @@
+import {assert} from 'chai';
+
+import {saveConnection} from '../../backend/persistent/Connections.js';
+import {saveSetting} from '../../backend/settings.js';
+
+import {
+    assertResponseStatus,
+    clearSettings,
+    closeTestServers,
+    createTestServers,
+    getResponseJson,
+    POST,
+    sqlConnections
+} from './utils.js';
+
+
+let servers;
+let connectionId;
+
+describe('Datastore Mock:', function () {
+    beforeEach(() => {
+        servers = createTestServers();
+        saveSetting('AUTH_ENABLED', false);
+        sqlConnections.mock = true;
+        connectionId = saveConnection(sqlConnections);
```

Comment:
You don't want to do this (the above two lines will overwrite `sample-storage/connections.yaml`).

I can't test this until Thursday, but I reckon it's enough to create a mock connections file in `mock-storage/connections.yaml` and set `STORAGE_PATH` accordingly:
```
saveSetting('STORAGE_PATH', 'mock-storage');
```

Reply:
I checked, and it does not. We are good here.

Reply:
How do you explain the failure in CircleCI?

Reply:
I don't have the rights to trigger a rebuild. Please, could you do it for me?

Reply:
To reproduce the error in CircleCI, you can run locally this:
```
NODE_ENV=test BABEL_DISABLE_CACHE=1 electron-mocha --bail --full-trace --timeout 90000 --compilers js:babel-register test/backend/datastores.mock.spec.js test/backend/routes.spec.js
```

Reply:
Note that `createTestServers` uses `sample-storage/connections.yaml`.

Reply:
> and set STORAGE_PATH accordingly:

Unfortunately: https://github.com/plotly/falcon-sql-client/blob/master/backend/settings.js#L171

```
        /*
         * STORAGE_PATH is the only variable that can't be saved in the
         * settings file because of recursion.
         */
```

I am going to remove `mock-connections.yaml`, and add a minor update to the original approach to get this working.

Reply:
There is no need to save `STORAGE_PATH`. Please, have a look at how the heroku app loads the sample connections.

Reply:
What is wrong with the current approach? It removes the use of an extra file/directory and does not leave any global effects.